rust/core-lib/src/view.rs

```diff
@@ -785,30 +773,9 @@ impl View {
         // the front-end, but perhaps not for async edits.
         self.drag_state = None;
 
-        // Update search highlights for changed regions
-        if self.search_string.is_some() {
-            self.valid_search = self.valid_search.apply_delta(delta);
-            let mut occurrences = self.occurrences.take().unwrap_or_else(Selection::new);
-
-            // invalidate occurrences around deletion positions
-            for DeltaRegion{ old_offset, new_offset, len } in delta.iter_deletions() {
-                self.valid_search.delete_range(new_offset, new_offset + len);
-                occurrences.delete_range(old_offset, old_offset + len, false);
-            }
-
-            occurrences = occurrences.apply_delta(delta, false, false);
-
-            // invalidate occurrences around insert positions
-            for DeltaRegion{ new_offset, len, .. } in delta.iter_inserts() {
-                self.valid_search.delete_range(new_offset, new_offset + len);
-                occurrences.delete_range(new_offset, new_offset + len, false);
-            }
-
-            self.occurrences = Some(occurrences);
-
-            // update find for the whole delta (is going to only update invalid regions)
-            let (iv, _) = delta.summary();
-            self.update_find(text, iv.start(), iv.end(), true, false);
+        // update only find highlights affected by change
+        for find in self.find.iter_mut() {
```

Comment:
Tiny style nit, could this be written as `for find in &mut self.find`? I tend to prefer that.

Reply:
Just want to say for the record that I prefer the original, I think this is exactly what `iter_mut()` is for. But I 100% don't mind raph's approach either. 😏